File: SOC/src/cc/game/soc/ui/MenuItem.java
package cc.game.soc.ui;

/**
 * Created by chriscaron on 2/27/18.
 */

public final class MenuItem {

    public interface Action {
        void onAction(MenuItem item, Object extra);
    };

    public final String title;
    public final String helpText;
    public final Action action;

    public MenuItem(String title, String helpText, Action action) {
        this.title = title;
        this.helpText = helpText;
        this.action = action;
    }

    @Override
    public final String toString() {
        return title;
    }
}

File: SOC/src/cc/game/soc/ui/MenuItem.kt
package cc.game.soc.ui

/**
 * Created by chriscaron on 2/27/18.
 */
class MenuItem(val title: String?, val helpText: String?, val action: Action?) {
	interface Action {
		fun onAction(item: MenuItem, extra: Any?)
	}

	override fun toString(): String {
		return title?:"null"
	}
}